pyhap/accessory_driver.py

```diff
@@ -493,23 +579,10 @@ def start(self):
             self.accessory.setup_message()
 
         # Start the accessory so it can do stuff.
-        if isinstance(self.accessory, AsyncAccessory):
-            self.accessory_task = self.loop.create_task(
-                self.accessory.run())
-        else:
-            self.accessory_task = self.loop.run_in_executor(
-                None, self.accessory.run)
-
-        logger.info("Starting event loop")
-        try:
-            self.loop.run_forever()
-        except KeyboardInterrupt:
-            pass
-        finally:
-            self.loop.close()
-            logger.info("Closed event loop.")
+        self.add_job(self.accessory.run)
```

Comment:
Can you please add a note to the changelog that old implementations of Accessory.run, which still use the `while True: sleep (x); do_stuff()` will exhaust the thread pool with never-ending tasks

Reply:
As long as it's not an async method that shouldn't be an issue, since it's basically the way it was before (just one never ending thread). However I agree that `run_at_interval` should be the preferred way. I'll add this.